spring-integration-core/src/main/java/org/springframework/integration/config/MessagingGatewayRegistrar.java

```diff
@@ -163,7 +164,10 @@ else if (StringUtils.hasText(asyncExecutor)) {
 
 		gatewayProxyBuilder.addConstructorArgValue(serviceInterface);
 
-		return new BeanDefinitionHolder(gatewayProxyBuilder.getBeanDefinition(), id);
+		AbstractBeanDefinition beanDefinition = gatewayProxyBuilder.getBeanDefinition();
+		beanDefinition.addMetadataAttribute(new BeanMetadataAttribute("factoryBeanObjectType", serviceInterface));
```

Comment:
On the Boot call yesterday, @philwebb mentioned some of the code moving into Spring Framework. It hasn't happened yet, AFAIK. Hopefully Phil can comment with the latest.